### PR TITLE
Initial development and testing

### PR DESCRIPTION
**Pull request by Chris-Schnaufer**
Testing changes and updated to base ODM (OpenDroneMap) image 0.9.1

Back reference to ticket: https://app.zenhub.com/workspaces/ua-ag-data-science-5a57a3198339f11ba1c85775/issues/terraref/drone-pipeline/96

Closing out PR on December 6 at end of day

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
+### Sample Docker Command line
+Below is a sample command line that shows how the OpenDroneMap Docker image could be run.
+An explanation of the command line options used follows.
+Be sure to read up on the [docker run](https://docs.docker.com/engine/reference/run/) command line for more information.
 
-## Extending the Template
-There are situations where this template won't be sufficient as a transformer for an environment.
-In these cases it's recommended that instead of forking this repo and making modifications, a new template repo is created with the expectation that the processing code will be a submodule to it.
-Scripts and/or instructions can then be provided on cloning this repo, specifying the submodule, and how to create a working transformer for the environment.
+```sh
```
Comment (dlebauer) [inline README.md]: FYI you can have markdown make code pretty if you tell it what language ... I made this change:

image

File: transformer_class.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+import os
+import argparse
+import datetime
+import logging
+import piexif
+
+import configuration
+
+# EXIF tags to look for
```
Comment (dlebauer) [inline transformer_class.py]:
```suggestion
# EXIF tags to look for, see https://www.exiv2.org/tags.html
```

36867 is linked to a tag called "Exif.Image.DateTimeOriginal" ... I don't want to get too picky on this but is ORIGIN a typo (did you mean EXIF_ORIGINAL_TIMESTAMP)  but are these consistent with the EXIF standard, i.e. is "The date and time when the original image data was generated. For a digital still camera the date and time the picture was taken are recorded." an accurate description of the EXIF_ORIGIN_TIMESTAMP variable?

File: worker.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+"""Worker script for OpenDroneMap transformer
+"""
+import os
+import yaml
+
+# Override pylint to allow our trickery in setting up ODM
```
Comment (dlebauer) [inline worker.py]: what is 'our trickery' ... please clarify...

File: README.md (hunk 1 above)
Comment (Chris-Schnaufer) [reply README.md]: Great!

File: opendronemap-overrides.txt.sample [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+#dtm: True      # Use this tag to build a DTM (Digital Terrain Model
+#dsm: True      # Use this tag to build a DSM (Digital Surface Model
+#dem-gapfill-steps: 4
+#dem-resolution: 0.1
+#dem-maxangle:20
+#dem-maxsd: 2.5
+#dem-approximate: False
+#dem-decimation: 1
+#dem-terrain-type: ComplexForest
+#orthophoto_resolution: 2.0 # Pixels/meter
```
[the remark this answers is not in the document]
Comment (Chris-Schnaufer) [reply opendronemap-overrides.txt.sample]: This file is directly from ODM as their parameter file. I'm making use of it to allow users to tweak what ODM does

To me it's not clear how this parameter works - that seems to have changed over time and hasn't been consistent.

File: settings.yaml [hunk 5]
```diff
@@ -5,16 +5,18 @@
 # or --force-ccd n will have to be set in the command line (if you need to)
 
 # This line is really important to set up properly
-project_path: '/code' # Example: '/home/user/ODMProjects
+project_path: '/' #DO NOT CHANGE THIS OR DOCKER WILL NOT WORK. It should be '/'
 
 # The rest of the settings will default to the values set unless you uncomment and change them
-#resize_to: 2400
+#resize_to: 2048
```
[the remark this answers is not in the document]
Comment (Chris-Schnaufer) [reply settings.yaml]: The overrides file allows users of the ODM transformer to override the default yaml settings. Both the yaml and overrides settings are updated for the latest ODM code

File: transformer_class.py (hunk 2 above)
Comment (Chris-Schnaufer) [reply transformer_class.py]: Please be picky. It looks like it's a typo - making change

File: worker.py (hunk 3 above)
Comment (Chris-Schnaufer) [reply worker.py]: Done

File: entrypoint.py [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    def parse_continue_result(result):
+        """Parses the result of calling transformer.check_continue and returns
+           the code and/or message
+        Arguments:
+            result: the result from calling transformer.check_continue
+        Return:
+            A tuple containing the result code and result message. One or both of these
+            values in the tuple may be None
+        Notes:
+            A string parameter is always return a result code will be None and message of None indicating
```
Comment (dlebauer) [inline entrypoint.py]: Unclear?

File: opendronemap-overrides.txt.sample [hunk 7]
```diff
@@ -0,0 +1,59 @@
+---
+# A list of global configuration variables
+# Uncomment lines as needed to edit default settings.
+# Note this only works for settings with default values.
```
[the remark this answers is not in the document]
Comment (Chris-Schnaufer) [reply opendronemap-overrides.txt.sample]: Added link to documentation

**Pull request merged**